Packet++/src/IPv6Extensions.cpp

```diff
@@ -207,15 +207,15 @@ IPv6RoutingHeader::IPv6RoutingHeader(uint8_t routingType, uint8_t segmentsLeft,
 
 	if (additionalRoutingDataLen > 0 && additionalRoutingData != NULL)
 	{
-		uint8_t* additionalDataPtr = (uint8_t*)(getDataPtr() + sizeof(ipv6_routing_header));
+		uint8_t* additionalDataPtr = getDataPtr() + sizeof(ipv6_routing_header);
 		memcpy(additionalDataPtr, additionalRoutingData, additionalRoutingDataLen);
 	}
 }
 
 uint8_t* IPv6RoutingHeader::getRoutingAdditionalData() const
 {
 	if (getExtensionLen() > sizeof(ipv6_routing_header))
-		return (uint8_t*)(getDataPtr() + sizeof(ipv6_routing_header));
+		return getDataPtr() + sizeof(ipv6_routing_header);
```

Comment:
why did you remove the casting?

Reply:
When a constant is added to a pointer the expression result will have the same type as pointer has. The `getDataPtr()` returns `uint8_t*`